### PR TITLE
Inform DevTools of commit priority level

### PR DESCRIPTION
**Pull request by bvaughn**
React now informs DevTools of the commit priority level (so the Profiler can display and potentially filter on the info).

### Demo
priority-Kapture 2019-05-15 at 15 21 57

### CODE REVIEW

**Comment by bvaughn** [comment]
🤔 Looks like importing `ReactFiberScheduler` from `ReactFiberDevToolsHook` causes `ReactIncrementalErrorReplay-test.internal` to fail with an invariant I don't understand yet:

> Invariant Violation: This module must be shimmed by a specific renderer.

File: packages/react-reconciler/src/ReactFiberDevToolsHook.js [hunk 1]
```diff
@@ -65,12 +52,39 @@ export function injectInternals(internals: Object): boolean {
...
-      hook.onCommitFiberRoot(rendererID, root),
-    );
-    onCommitFiberUnmount = catchErrors(fiber =>
-      hook.onCommitFiberUnmount(rendererID, fiber),
-    );
+    onCommitFiberRoot = (root, expirationTime) => {
+      const currentTime = requestCurrentTime();
+      const priorityLevel = inferPriorityFromExpirationTime(
+        currentTime,
+        expirationTime,
```
Comment (gaearon) [inline packages/react-reconciler/src/ReactFiberDevToolsHook.js]: Should these be DEV/PROFILE only?

Comment (bvaughn) [reply packages/react-reconciler/src/ReactFiberDevToolsHook.js]: Interesting. Yeah probably wouldn't hurt!

File: packages/react-reconciler/src/ReactFiberDevToolsHook.js [hunk 2]
```diff
@@ -65,12 +53,43 @@ export function injectInternals(internals: Object): boolean {
...
-      hook.onCommitFiberUnmount(rendererID, fiber),
-    );
+    onCommitFiberRoot = (root, expirationTime) => {
+      try {
+        if (enableProfilerTimer) {
+          const currentTime = requestCurrentTime();
+          const priorityLevel = inferPriorityFromExpirationTime(
+            currentTime,
+            expirationTime,
+          );
```
Comment (bvaughn) [inline packages/react-reconciler/src/ReactFiberDevToolsHook.js]: Andrew mentioned maybe we could use [`getCurrentPriorityLevel()`](https://github.com/facebook/react/blob/9c6de716d028f17736d0892d8a3d8f3ac2cb62bd/packages/react-reconciler/src/SchedulerWithReactIntegration.js#L87-L102) here rather than inferring the priority level. I don't _think_ the priority level a Fiber was scheduled with will necessarily match the current priority level though. (Maybe I misunderstood his suggestion.)

File: packages/react-reconciler/src/__tests__/ReactIncrementalErrorReplay-test.internal.js [hunk 3]
```diff
@@ -11,6 +11,10 @@
 
 describe('ReactIncrementalErrorReplay-test', () => {
   it('copies all keys when stashing potentially failing work', () => {
+    // Include a renderer to ensure host config files are properly shimmed.
+    // Otherwise transient imports may cause an invariant.
+    require('react-dom');
```
Comment (gaearon) [inline packages/react-reconciler/src/__tests__/ReactIncrementalErrorReplay-test.internal.js]: I landed another PR which needed this fix so you should be able to revert this file

Comment (bvaughn) [reply packages/react-reconciler/src/__tests__/ReactIncrementalErrorReplay-test.internal.js]: Oh, cool thanks.

File: packages/react-reconciler/src/ReactFiberDevToolsHook.js (hunk 2 above)
Comment (acdlite) [reply packages/react-reconciler/src/ReactFiberDevToolsHook.js]: If you call `getCurrentPriorityLevel` inside `commitRoot` (not `commitRootImpl`, since that's always run at `ImmediatePriority` — it has to be the wrapper) then it will match the priority of the commit, which is not necessarily the same thing as the priority at which something was scheduled because multiple priorities can get batched.

This approach seems fine, though.

It's possible we may eventually track the priority on the update object.
